Source/Core/Core/PowerPC/GDBStub.cpp

```diff
@@ -826,123 +776,87 @@ void gdb_init(u32 port)
 {
   s_socket_context.emplace();
-  memset(bp_x, 0, sizeof bp_x);
-  memset(bp_r, 0, sizeof bp_r);
-  memset(bp_w, 0, sizeof bp_w);
-  memset(bp_a, 0, sizeof bp_a);
 
-  tmpsock = socket(domain, SOCK_STREAM, 0);
-  if (tmpsock == -1)
+  s_tmpsock = socket(domain, SOCK_STREAM, 0);
+  if (s_tmpsock == -1)
     ERROR_LOG_FMT(GDB_STUB, "Failed to create gdb socket");
 
   int on = 1;
-  if (setsockopt(tmpsock, SOL_SOCKET, SO_REUSEADDR, (const char*)&on, sizeof on) < 0)
+  if (setsockopt(s_tmpsock, SOL_SOCKET, SO_REUSEADDR, (const char*)&on, sizeof on) < 0)
     ERROR_LOG_FMT(GDB_STUB, "Failed to setsockopt");
 
-  if (bind(tmpsock, server_addr, server_addrlen) < 0)
+  if (bind(s_tmpsock, server_addr, server_addrlen) < 0)
     ERROR_LOG_FMT(GDB_STUB, "Failed to bind gdb socket");
 
-  if (listen(tmpsock, 1) < 0)
+  if (listen(s_tmpsock, 1) < 0)
     ERROR_LOG_FMT(GDB_STUB, "Failed to listen to gdb socket");
 
   INFO_LOG_FMT(GDB_STUB, "Waiting for gdb to connect...");
 
-  sock = accept(tmpsock, client_addr, client_addrlen);
-  if (sock < 0)
+  s_sock = accept(s_tmpsock, client_addr, client_addrlen);
+  if (s_sock < 0)
     ERROR_LOG_FMT(GDB_STUB, "Failed to accept gdb client");
   INFO_LOG_FMT(GDB_STUB, "Client connected.");
 
 #ifdef _WIN32
   closesocket(tmpsock);
 #else
-  close(tmpsock);
+  close(s_tmpsock);
 #endif
-  tmpsock = -1;
+  s_tmpsock = -1;
+
+  s_update_event = CoreTiming::RegisterEvent("GDBStubUpdate", UpdateCallback);
+  CoreTiming::ScheduleEvent(GDB_UPDATE_CYCLES, s_update_event);
+  s_has_control = true;
 }
 
-void gdb_deinit()
+void Deinit()
 {
-  if (tmpsock != -1)
+  if (s_tmpsock != -1)
   {
-    shutdown(tmpsock, SHUT_RDWR);
-    tmpsock = -1;
+    shutdown(s_tmpsock, SHUT_RDWR);
+    s_tmpsock = -1;
   }
-  if (sock != -1)
+  if (s_sock != -1)
   {
-    shutdown(sock, SHUT_RDWR);
-    sock = -1;
+    shutdown(s_sock, SHUT_RDWR);
+    s_sock = -1;
   }
 
   s_socket_context.reset();
+  s_has_control = false;
 }
 
-bool gdb_active()
+bool IsActive()
 {
-  return tmpsock != -1 || sock != -1;
+  return s_tmpsock != -1 || s_sock != -1;
 }
 
-int gdb_signal(u32 s)
+bool HasControl()
 {
-  if (sock == -1)
-    return 1;
-
-  sig = s;
-
-  if (send_signal)
-  {
-    gdb_handle_signal();
-    send_signal = 0;
-  }
-
-  return 0;
+  return s_has_control;
 }
 
-int gdb_bp_x(u32 addr)
+void TakeControl()
 {
-  if (sock == -1)
-    return 0;
-
-  if (step_break)
-  {
-    step_break = 0;
-
-    DEBUG_LOG_FMT(GDB_STUB, "Step was successful.");
-    return 1;
-  }
-
-  return gdb_bp_check(addr, GDB_BP_TYPE_X);
+  s_has_control = true;
 }
 
-int gdb_bp_r(u32 addr)
+void SendSignal(u32 signal)
 {
-  if (sock == -1)
-    return 0;
-
-  return gdb_bp_check(addr, GDB_BP_TYPE_R);
-}
-
-int gdb_bp_w(u32 addr)
-{
-  if (sock == -1)
-    return 0;
-
-  return gdb_bp_check(addr, GDB_BP_TYPE_W);
-}
-
-int gdb_bp_a(u32 addr)
-{
-  if (sock == -1)
-    return 0;
-
-  return gdb_bp_check(addr, GDB_BP_TYPE_A);
+  char bfr[128];
```

Comment:
```suggestion
  char bfr[128] = {};
```

The memset below can be removed